### PR TITLE
Fix build warning with memset value being too large

### LINKED ISSUE #83342: GLES3: Texture's memset exceeds maximum object size
### Godot version

4.2.beta [a574c02]

### System information

Arch Linux 6.5.6-zen2-1-zen (64-bit)

### Issue description

When the compiler tries to compiles `drivers/gles3/texture_storage.cpp`, it only gets one warning saying that `memset` specified bound will be exceeds maximum object size

### Steps to reproduce

1. Clone (if you haven't)
2. Compile it
3. Look out for something like `warning: 'void* memset(void*, int, size_t)' specified bound between 18446744065119617024 and 18446744073709551612 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]`
3.5. If isn't then, try different `target` like `target=template_release` and look again, repeat until either it's there or somehow gone.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by griffinkh**
Should resolve issue #83342

* *Bugsquad edit, closes: #83342*

### CODE REVIEW

File: drivers/gles3/storage/texture_storage.cpp [hunk 1]
```diff
@@ -1475,7 +1475,7 @@ void TextureStorage::update_texture_atlas() {
 		//generate atlas
 		Vector<TextureAtlas::SortItem> itemsv;
 		itemsv.resize(texture_atlas.textures.size());
-		int base_size = 8;
+		unsigned int base_size = 8;
```
Comment (clayjohn) [inline drivers/gles3/storage/texture_storage.cpp:1478]: We use `uint32_t` instead of `unsigned int`
```suggestion
		uint32_t base_size = 8;
```

**Review by clayjohn — CHANGES_REQUESTED** [review]
Instead of using `unsigned int` this should use `uint32_t`, otherwise this seems great!

File: drivers/gles3/storage/texture_storage.cpp (hunk 1 above)
Comment (griffinkh) [reply drivers/gles3/storage/texture_storage.cpp]: thanks! fixed

**Comment by griffinkh** [comment]
change is made, thanks!

**Review by clayjohn — APPROVED** [review]
Looks great! Thanks for the quick update

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by akien-mga** [comment]
Cherry-picked for `4.2`, `4.1`, and `4.0`.

**Pull request merged**